dodo.py

```diff
@@ -1706,11 +1756,88 @@ def _ensure_resolutions(app_name):
                 for k, v in sorted(app["resolutions"].items(), key=lambda item: item[0])
             }
 
-            # Write the package.json back to disk.
-            app_json.write_text(json.dumps(app, indent=2) + "\n", **C.ENC)
+            new_raw = json.dumps(app, indent=2)
+
+            if raw.strip() == new_raw.strip():
+                return True
+
+            print(
+                "\n".join(
+                    difflib.context_diff(
+                        raw.splitlines(), new_raw.splitlines(), "old", "new"
+                    )
+                )
+            )
+
+            if C.CI and not C.CHECKING_RELEASE:
```

Comment:
Maybe we still need to run this in the check release workflow to make sure resolutions have been updated and not leave a git dirty state.

Reply:
Hm... i guess i though that check-release _should_ be allowed to changes things, while "plain old lint" should just fail hard if a new package gets added, for example (which is what bit me)